src/main/java/botmanager/Storage.java

```diff
@@ -0,0 +1,80 @@
+package botmanager;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.Scanner;
+
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.TaskList;
+import task.Todo;
+
+/**
+ * Class to handle reading and writing from data file.
+ *
+ * @author dwsc37
+ */
+public class Storage {
+    private static final String TASK_FILE_PATH = "data/tasks.txt";
+    private static final String DIRECTORY_PATH = "data";
+
+    /**
+     * Reads tasks from data file and adds them to the given <code>taskList</code>.
+     *
+     * @param taskList Task list to add tasks to.
+     * @throws FileNotFoundException If data file is not found.
+     */
+    public void loadTaskList(TaskList taskList) throws FileNotFoundException {
+        File f = new File(TASK_FILE_PATH);
+        Scanner s = new Scanner(f);
+        while (s.hasNextLine()) {
+            String data = s.nextLine();
+            taskList.addTask(fromData(data));
+        }
+    }
+
+    private Task fromData(String data) {
+        String[] args = data.split(" \\| ");
+        return switch (args[0]) {
```

Comment:
Great job on this! Many users make the mistake of indenting the case statements.